src/App.tsx

```diff
@@ -1,15 +1,49 @@
 import React from 'react';
+import posts from './api/posts';
+import comments from './api/comments';
+import users from './api/users';
+import { PostList } from './components/PostList';
+import {
+  PostsFromServer,
+  CommentsFromServer,
+  UsersFromServer,
+  GeneralPosts,
+} from './react-app-env';
 
 import './App.scss';
 
-// import posts from './api/posts';
-// import comments from './api/comments';
-// import users from './api/users';
+const App: React.FC = () => {
+  const getNewPostList = (
+    postsList: PostsFromServer[],
+    commentsList: CommentsFromServer[],
+    usersList: UsersFromServer[],
+  ): GeneralPosts[] => {
+    return postsList.map(post => ({
+      ...post,
+      user: usersList
+        .find(user => post.userId === user.id)
+        || null,
+      comments: commentsList
```

Comment:
Method filter always returns an array, so null never be the value of comments field